mono/metadata/threads.c

```diff
@@ -2514,7 +2557,8 @@ ves_icall_System_Threading_Thread_ResetAbort (MonoThread *this_obj)
 
 	if (!was_aborting) {
 		const char *msg = "Unable to reset abort because no abort was requested";
-		mono_set_pending_exception (mono_get_exception_thread_state (msg));
+		mono_set_pending_exception_handle (mono_exception_new_thread_state (msg, error));
+		mono_error_assert_ok (error);
```

Comment:
This assert is confusing but probably ok. It is asserting that creating the exception succeeded. i.e. not low memory. The actual assert created and set is in the thread.